webrtc.html

```diff
@@ -7699,7 +7699,37 @@ <h3><dfn>RTCDtlsTransport</dfn> Interface</h3>
           at <var>transport</var>.</p>
         </li>
       </ol>
-
+      <p>When the underlying DTLS transport experiences an error, such as
+        a certificate validation failure, or a fatal alert (see RFC 5246
+        section 7.2), the user agent MUST queue a task that runs the following
+        steps:
+      </p>
+      <ol>
+        <li>
+          <p>Let <var>transport</var> be the <code>
+          <a>RTCDtlsTransport</a></code> object to receive the state update.</p>
+        </li>
+        <li>
+          <p>Set <var>transport</var>'s <a>[[\DtlsTransportState]]</a> slot
+            to <code>failed</code>.</p>
+        </li>
+        <li>
+          <p>
+            <a>Fire an event</a> named
+            <code><a data-lt="RTCDtlsTransport error">error</a></code>
+            using the <a>RTCErrorEvent</a> interface with its errorDetail
+            attribute set to either "dtls-failure" or "fingerprint-failure",
+            as appropriate, and
+            other fields set as described under the <a>RTCErrorDetailType</a>
+            enum description, at <var>transport</var>.
+          </p>
+        </li>
+        <li>
+          <p><a>Fire an event</a> named <code><a data-lt="RTCDtlsTransport state change">statechange</a></code>
+          at <var>transport</var>.</p>
```

Comment:
It's good to get the order of these things well-defined. Note that we already have "When the underlying DTLS transport needs to update the state of the corresponding RTCDtlsTransport object, the user agent MUST queue a task that runs the following steps" though so a stupid reading of the spec might attempt to fire the event twice.

Perhaps the other "statechange" steps should be updated to say "needs to update the state for any other reason than experiencing an error"?

Reply:
Agreed. Will swap the two sections too.

Reply:
Don't fire the event if you were already failed.